tiff-2-png-converter.py

```diff
@@ -7,100 +7,249 @@
+            for i in range(3):
+                current_channel = image_channels[i]
+                if current_channel[z] is not None:
+                    current_image_channels[i] = current_channel[z]
+                    if arg_values["equalize_histogram"]:
+                        current_image_channels[i] = arg_values[
+                            "equalize_histogram_method"
+                        ](current_image_channels[i])
+                else:
+                    current_image_channels[i] = zero_image
+
+            merged_image = cv2.merge(current_image_channels)
 
-        if args.gamma:
-            TiffProcessor.gamma = True
-            TiffProcessor.gamma_val = args.gamma
-        if args.num_imgs:
-            TiffProcessor.num_imgs = int(args.num_imgs)
-
-    def convert_to_png(self):
-        list_files, width, heigth = self.list_folder_files(self.source)
-        if self.num_imgs is None:
-            self.num_images = len(list_files)
-        image_r = [None] * self.num_imgs
-        image_g = [None] * self.num_imgs
-        image_b = [None] * self.num_imgs
-        images = [None] * self.num_imgs
-        bits_per_sample = ""
-        print_progress_bar(
-            0, self.num_imgs, prefix="grouping channels:", suffix="Complete", length=50
-        )
-        for i in range(len(list_files)):
+            if arg_values["gamma"] is not None:
+                float_gamma = float(arg_values["gamma"])
+                merged_image = iph.gamma_correction(merged_image, float_gamma)
+            rgb_images[z] = merged_image
+
+
+def build_image_sequence(volume, width, height, dtype):
+    z_slices = arg_values["nimgs"] + 1
+    dim = math.ceil(math.sqrt(z_slices))
+    max_res = math.floor(4096 / dim)
+    downscale = math.ceil(max(width, height) / max_res)
+    if arg_values["verbose"]:
+        logging.info("resolution_image_before " + str(width) + " " + str(height))
+        logging.info("downscale " + str(downscale))
+
+    resolution_image = (int(width / downscale), int(height / downscale))
+    count = 0
+    image_out = np.zeros(())
+
+    with typer.progressbar(range(dim), label="Mapping slices to 2D") as progress:
+        for i in progress:
+            imgs_row = np.zeros((), dtype=dtype)
+            for j in range(dim):
+                tmp_img = np.zeros((resolution_image), dtype=dtype)
+                img_index = (0, count)[
+                    count < len(volume) and count <= arg_values["nimgs"]
+                ]
+                img = volume[img_index]
+
+                if img is not None:
+                    tmp_img = cv2.resize(img, resolution_image)
+
+                if img_index == 0:
+                    tmp_img[:, :] = 0
+
+                if j == 0:
+                    imgs_row = np.copy(tmp_img)
+                else:
+                    imgs_row = cv2.hconcat([imgs_row, tmp_img])
+                count = count + 1
+
+            if i == 0:
```

Comment:
And in here, I'm wondering if you can use array indexing instead of vcats and hcats so you don't have to dynamically grow the array